core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.java

```diff
@@ -1342,89 +1326,15 @@ private void searchForTitle(String title, boolean openInNewTab) {
     }
   }
 
-  @Override
-  public void onActivityResult(int requestCode, int resultCode, Intent data) {
-    Log.i(TAG_KIWIX, "Intent data: " + data);
-    switch (requestCode) {
-      case MainMenuKt.REQUEST_FILE_SEARCH:
-        if (resultCode == RESULT_OK) {
-          boolean wasFromTabSwitcher = isInTabSwitcher();
-          hideTabSwitcher();
-          String title =
-            data.getStringExtra(TAG_FILE_SEARCHED).replace("<b>", "").replace("</b>", "");
-          boolean isSearchInText =
-            data.getBooleanExtra(SearchInPreviousScreen.EXTRA_SEARCH_IN_TEXT, false);
-          if (isSearchInText) {
-            //if the search is localized trigger find in page UI.
-            KiwixWebView webView = getCurrentWebView();
-            compatCallback.setActive();
-            compatCallback.setWebView(webView);
-            ((AppCompatActivity) getActivity()).startSupportActionMode(compatCallback);
-            compatCallback.setText(title);
-            compatCallback.findAll();
-            compatCallback.showSoftInput();
-          } else {
-            boolean openInNewTab = wasFromTabSwitcher ||
-              data.getBooleanExtra(TAG_FILE_SEARCHED_NEW_TAB, false);
-            searchForTitle(title, openInNewTab);
-          }
-        } else if (resultCode == RESULT_CANCELED) {
-          Log.w(TAG_KIWIX, "Search cancelled or exited");
-        } else {
-          Log.w(TAG_KIWIX, "Unhandled search failure");
-          Toast.makeText(getActivity(), R.string.search_error, Toast.LENGTH_SHORT).show();
-        }
-        break;
-      case REQUEST_PREFERENCES:
-        hideTabSwitcher();
-        if (resultCode == RESULT_RESTART) {
-          getActivity().recreate();
-        }
-        if (resultCode == RESULT_HISTORY_CLEARED) {
-          webViewList.clear();
-          newMainPageTab();
-          tabsAdapter.notifyDataSetChanged();
-        }
-        loadPrefs();
-        break;
-
-      case BOOKMARK_CHOSEN_REQUEST:
-      case REQUEST_FILE_SELECT:
```

Comment:
We can still open bookmarks and history items, yeah?

Reply:
Yes, this fragment does not use any activity for result and bookmarks and history is handled by navigation arguments.